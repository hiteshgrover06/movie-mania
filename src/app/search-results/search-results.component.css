:host {
  width: 100vw;
}

.search-results {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
}


/* For mobile */
.show-item {
  width: inherit;
  margin: 0.5rem;
}

.search-results img {
  width: 100%;
  height: 90%;
}

/* For tablet */
@media only screen and (min-width: 600px) { 
  .search-results img {
    width: initial;
    height: max-content;
  }
}

/* For desktop */
@media only screen and (min-width: 768px) {
  .show-item {
    width: min-content;
    height: min-content;
  }
  .search-results img {
    width: inherit;
    height: inherit;
  }
}
