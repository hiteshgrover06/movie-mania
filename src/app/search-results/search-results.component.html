<div class="search-conatiner">
  <div class="section-title">Search results for "{{ query }}"</div>

  <p
    *ngIf="(searchResults$ | async)?.length === 0"
    class="section-title warning"
  >
    No results found , please try a diffrent query.
  </p>

  <!-- Search Results section -->
  <ng-container *ngIf="searchResults$ | async as results; else loading">
    <div class="search-results">
      <ng-container *ngFor="let item of results">
        <a class="show-item" (click)="goToDetailsPage($event, item.show.id)">
          <img
            width="220"
            height="240"
            class="show-image"
            src="{{
              item.show.image?.medium ||
                item.show.image?.original ||
                '../../assets/images/no-preview.png'
            }}"
            alt="show banner"
          />
          <div class="">
            <ul>
              <li><b>Title - </b> {{ item.show?.name }}</li>
              <li>
                <b>Language - </b>
                {{ item.show?.language }}
              </li>
              <li>
                <b>Rating - </b>
                {{ detail?.rating?.average }}
              </li>
            </ul>
          </div>
        </a>
      </ng-container>
    </div>
  </ng-container>

  <ng-template #loading>
    <div class="category">
      <div class="category-shows">
        <ng-container *ngFor="let show of [1, 2, 3, 4]">
          <div class="show-item">
            <img
              width="500"
              height="240"
              class="show-image skeleton"
              src=""
              alt=""
            />
          </div>
        </ng-container>
      </div>
    </div>
  </ng-template>
</div>
