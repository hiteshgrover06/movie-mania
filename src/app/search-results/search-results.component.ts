import { Component, OnInit } from "@angular/core";
import { ActivatedRoute, Router } from "@angular/router";
import { Observable } from "rxjs/internal/Observable";
import {
  SearchDetail,
  SearchShowsService,
} from "src/services/search-shows.service";

@Component({
  selector: "app-search-results",
  templateUrl: "./search-results.component.html",
  styleUrls: ["./search-results.component.css"],
})
export class SearchResultsComponent implements OnInit {
  constructor(
    private readonly searchService: SearchShowsService,
    private readonly activeRoute: ActivatedRoute,
    private readonly router: Router
  ) {}
  searchResults$: Observable<SearchDetail[]>;
  query: string;

  ngOnInit() {
    this.activeRoute.params.subscribe((routeParams) => {
      this.query = routeParams && routeParams[`query`];
      if (this.query) {
        this.searchResults$ = null;
        // To simulate slow network or time consuming calls
        setTimeout(() => {
          this.searchResults$ = this.searchService.searchShows(
            `${this.query.trim()}`
          );
        }, 2000);
      }
    });
  }

  // navigate to the selected show detail
  goToDetailsPage($event: Event, showId: number) {
    $event.preventDefault();
    this.router.navigateByUrl(`/show/${showId}`);
  }
}
