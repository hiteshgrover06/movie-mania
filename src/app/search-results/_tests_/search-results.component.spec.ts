import { HttpClient, HttpHandler } from "@angular/common/http";
import { async, ComponentFixture, TestBed } from "@angular/core/testing";
import { RouterTestingModule } from "@angular/router/testing";
import { SearchShowsService } from "src/services/car-details.service";

import { SearchResultsComponent } from "../search-results.component";

describe("SearchResultsComponent", () => {
  let component: SearchResultsComponent;
  let fixture: ComponentFixture<SearchResultsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [SearchResultsComponent],
      providers: [SearchShowsService, HttpClient, HttpHandler],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchResultsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it("should create", () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it("should show search results", () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
