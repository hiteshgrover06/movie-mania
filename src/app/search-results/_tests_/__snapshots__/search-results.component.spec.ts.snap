// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsComponent should create 1`] = `
<div
  id="root0"
>
  <div
    class="search-conatiner"
  >
    <div
      class="section-title"
    >
      Search results for ""
    </div>
    
    
    <div
      class="category"
    >
      <div
        class="category-shows"
      >
        
        
        <div
          class="show-image skeleton"
        />
        
        <div
          class="show-image skeleton"
        />
        
        <div
          class="show-image skeleton"
        />
        
        <div
          class="show-image skeleton"
        />
        
        <div
          class="show-image skeleton"
        />
        
        <div
          class="show-image skeleton"
        />
        
        <div
          class="show-image skeleton"
        />
      </div>
    </div>
    
  </div>
</div>
`;

exports[`SearchResultsComponent should show search results 1`] = `
<div
  id="root1"
>
  <div
    class="search-conatiner"
  >
    <div
      class="section-title"
    >
      Search results for "query"
    </div>
    
    
    
    <div
      class="search-results"
    >
      
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/20/51263.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Samurai Girl
            </li>
            <li>
              <b>
                Language - 
              </b>
               English 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 6.5
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/36/90559.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Red Data Girl
            </li>
            <li>
              <b>
                Language - 
              </b>
               Japanese 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="../../assets/images/no-preview.png"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Gata Salvaje
            </li>
            <li>
              <b>
                Language - 
              </b>
               Spanish 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/172/431057.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Blind Date
            </li>
            <li>
              <b>
                Language - 
              </b>
               English 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/159/399537.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Dino Dana
            </li>
            <li>
              <b>
                Language - 
              </b>
               English 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/27/68631.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Dama til
            </li>
            <li>
              <b>
                Language - 
              </b>
               Norwegian 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/86/216264.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Мата Хари
            </li>
            <li>
              <b>
                Language - 
              </b>
               Russian 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/248/621949.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               UFO Baby
            </li>
            <li>
              <b>
                Language - 
              </b>
               Japanese 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/110/277293.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               Dara and Ed's Road to Mandalay
            </li>
            <li>
              <b>
                Language - 
              </b>
               English 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
      
      <a
        class="show-item"
      >
        <img
          alt="show banner"
          class="show-image"
          height="240"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/271/677662.jpg"
          width="220"
        />
        <div>
          <ul>
            <li>
              <b>
                Title - 
              </b>
               The School Nurse Files
            </li>
            <li>
              <b>
                Language - 
              </b>
               Korean 
            </li>
            <li>
              <span>
                <i
                  class="rating material-icons"
                >
                  star
                </i>
                 n/a
              </span>
            </li>
          </ul>
        </div>
      </a>
    </div>
    
  </div>
</div>
`;
